snews_pt/__main__.py

```diff
@@ -88,6 +89,24 @@ def publish(ctx, file, firedrill, verbose):
                     snews_messages = messages.create_messages(**json_data)
                     for message in snews_messages:
                         publisher.add_message(message)
+                        json_detector_name = message.detector_name
+                        if json_detector_name != env_detector_name:
+                            click.secho(
+                                f"{click.style('Warning:',bg = 'red')} "
+                                f"Detector name in JSON file ({click.style(
+                                    json_detector_name, bold = True)}) "
+                                f"does not match the environment detector name "
+                                f"{click.style(env_detector_name, bold = True)}",
+                                fg="red", color="black"
+                            )
+                            click.secho(
+                                f"{click.style('Warning:',bg = 'red')} "
+                                f"Using environment detector name "
+                                f"{click.style(env_detector_name, bold = True)} "
+                                f"for this message. "
+                                f"To change the detector name, use the `snews_pt set-name` command.",
+                            )
+                        message.detector_name = env_detector_name #force detector name from env
```

Comment:
Looking back, the change in name is after the message is put on the publisher queue! Would do nothing...